src/Libero/Sniffs/Types/IterableTypesSniff.php

```diff
@@ -0,0 +1,358 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\CodingStandard\Libero\Sniffs\Types;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHPStan\PhpDocParser\Ast\PhpDoc\ParamTagValueNode;
+use PHPStan\PhpDocParser\Ast\PhpDoc\PhpDocNode;
+use PHPStan\PhpDocParser\Ast\PhpDoc\ReturnTagValueNode;
+use PHPStan\PhpDocParser\Ast\PhpDoc\VarTagValueNode;
+use PHPStan\PhpDocParser\Ast\Type\ArrayTypeNode;
+use PHPStan\PhpDocParser\Ast\Type\GenericTypeNode;
+use PHPStan\PhpDocParser\Ast\Type\IdentifierTypeNode;
+use PHPStan\PhpDocParser\Ast\Type\IntersectionTypeNode;
+use PHPStan\PhpDocParser\Ast\Type\NullableTypeNode;
+use PHPStan\PhpDocParser\Ast\Type\TypeNode;
+use PHPStan\PhpDocParser\Ast\Type\UnionTypeNode;
+use PHPStan\PhpDocParser\Lexer\Lexer;
+use PHPStan\PhpDocParser\Parser\ConstExprParser;
+use PHPStan\PhpDocParser\Parser\PhpDocParser;
+use PHPStan\PhpDocParser\Parser\TokenIterator;
+use PHPStan\PhpDocParser\Parser\TypeParser;
+use SlevomatCodingStandard\Helpers\AnnotationHelper;
+use SlevomatCodingStandard\Helpers\DocCommentHelper;
+use SlevomatCodingStandard\Helpers\FunctionHelper;
+use SlevomatCodingStandard\Helpers\PropertyHelper;
+use SlevomatCodingStandard\Helpers\ReturnTypeHint;
+use Traversable;
+use function array_map;
+use function array_reduce;
+use function implode;
+use function in_array;
+use function is_a;
+use function is_string;
+use function sprintf;
+use function strpos;
+use const T_FUNCTION;
+use const T_VARIABLE;
+
+final class IterableTypesSniff implements Sniff
+{
+    public const CODE_MISSING_PROPERTY_ITERABLE_TYPE_HINT = 'MissingPropertyIterableType';
+    public const CODE_MISSING_FUNCTION_ITERABLE_ARGUMENT_TYPE_HINT = 'MissingIterableArgumentType';
+    public const CODE_MISSING_FUNCTION_ITERABLE_RETURN_TYPE_HINT = 'MissingIterableReturnType';
+    public const CODE_PROPERTY_ITERABLE_TYPE_HINT_GENERICS = 'PropertyIterableTypeGenerics';
+    public const CODE_ARGUMENT_ITERABLE_TYPE_HINT_GENERICS = 'ArgumentIterableTypeGenerics';
+    public const CODE_METHOD_ITERABLE_RETURN_TYPE_HINT_GENERICS = 'MethodIterableReturnTypeGenerics';
+    public const SNIFF_NAME = 'Libero.Types.IterableTypes';
+
+    /** @var Lexer */
+    private $lexer;
+
+    /** @var PhpDocParser */
+    private $parser;
+
+    public function __construct()
+    {
+        $this->lexer = new Lexer();
+        $this->parser = new PhpDocParser(new TypeParser(), new ConstExprParser());
+    }
+
+    /**
+     * @return array<int>
+     */
+    public function register() : array
+    {
+        return [
+            T_VARIABLE,
+            T_FUNCTION,
+        ];
+    }
+
+    // phpcs:ignore SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingParameterTypeHint
```

Comment:
This is annoying. `$pointer` doesn't have a type hint; adding a `@param` here sees it deleted by the fixer. 🤷‍♂️